boards/arm/mps2_an521/CMakeLists.txt

```diff
@@ -1,9 +1,85 @@
 #
-# Copyright (c) 2019 Linaro Limited
+# Copyright (c) 2019,2020 Linaro Limited
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 
 zephyr_library()
 zephyr_library_sources(pinmux.c)
 zephyr_library_include_directories(${ZEPHYR_BASE}/drivers)
+
+if (CONFIG_BUILD_WITH_TFM)
+	# Set default image versions if not defined elsewhere
+	if (NOT DEFINED TFM_IMAGE_VERSION_S)
+		set(TFM_IMAGE_VERSION_S 0.0.0+0)
+	endif()
+
+	if (NOT DEFINED TFM_IMAGE_VERSION_NS)
+		set(TFM_IMAGE_VERSION_NS 0.0.0+0)
+	endif()
+
+	set(PREPROCESSED_FILE "${CMAKE_BINARY_DIR}/tfm/image_macros_preprocessed")
+	set(TFM_MCUBOOT_DIR "${ZEPHYR_BASE}/../modules/tee/tfm/trusted-firmware-m/bl2/ext/mcuboot")
+
+	# Configure which format (full or hash) to include the public key in
+	# the image manifest
+	set(TFM_PUBLIC_KEY_FORMAT "full")
+
+	# Set srec_cat binary name
+	find_program(SREC_CAT srec_cat)
```

Comment:
Not enough.
You need to safe guard, and maybe even inform the users that there are missing dependencies:
```
**[INFO]** Image version number set to 0.0.0+1
**[INFO]** Image version number set to 0.0.0+2
/bin/sh: 1: SREC_CAT-NOTFOUND: not found
ninja: build stopped: subcommand failed.
```

Reply:
```
# Set srec_cat binary name
find_program(SREC_CAT srec_cat)
if(NOT SREC_CAT)
    message(FATAL_ERROR "'srec_cat' not found. Please install it, or add it to $PATH.")
endif()
```
Is this check sufficient, in your opinion?

Reply:
will not work:
>  If nothing is found, the result will be `<VAR>-NOTFOUND`
https://cmake.org/cmake/help/v3.17/command/find_program.html

You need to check as:
```
# Set srec_cat binary name
find_program(SREC_CAT srec_cat)
if(${SREC_CAT} STREQUAL SREC_CAT-NOTFOUND)
    message(FATAL_ERROR "'srec_cat' not found. Please install it, or add it to $PATH.")
endif()
```